plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/TransformPom.java

```diff
@@ -0,0 +1,58 @@
+package org.jenkins.tools.test.hook;
+
+import org.jenkins.tools.test.model.MavenCoordinates;
+import org.jenkins.tools.test.model.MavenPom;
+import org.jenkins.tools.test.model.PomData;
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeExecution;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class TransformPom extends PluginCompatTesterHookBeforeExecution {
```

Comment:
This is a transformation that happens in the plugin-compat-tester today.  I moved it out to a hook since it fit this check-action format.